selfcal_helpers.py

```diff
@@ -2987,11 +2983,16 @@ def analyze_inf_EB_flagging(selfcal_library,band,spwlist,gaintable,vis,target,sp
    for i in range(len(spwlist)):
       eff_bws[i]=selfcal_library[target][band][vis]['per_spw_stats'][keylist[i]]['effective_bandwidth']
       total_bws[i]=selfcal_library[target][band][vis]['per_spw_stats'][keylist[i]]['bandwidth']
-   minimum_flagged_ants_per_spw=np.min(nflags)/2.0
-   minimum_flagged_ants_spwcomb=np.min(nflags_spwcomb)/2.0 # account for the fact that some antennas might be completely flagged and give 
+   minimum_flagged_ants_per_spw=np.min(nflags)
+   minimum_flagged_ants_spwcomb=np.min(nflags_spwcomb) # account for the fact that some antennas might be completely flagged and give 
                                                            # the impression of a lot of flagging
-   maximum_flagged_ants_per_spw=np.max(nflags)/2.0
-   delta_nflags=np.array(nflags)/2.0-minimum_flagged_ants_spwcomb #minimum_flagged_ants_per_spw
+   maximum_flagged_ants_per_spw=np.max(nflags)
+   delta_nflags=np.array(nflags)-minimum_flagged_ants_spwcomb #minimum_flagged_ants_per_spw
+
+   # if more than two antennas (in 45 antenna array) are fully flagged relative to the combinespw results, fallback to combinespw
+   max_flagged_ants_combspw=(2./45)*np.max(np.array(nflags)+np.array(nunflagged))
+   # if only a single (or few) spw(s) has flagging, allow at most this number of antennas to be flagged before mapping
+   max_flagged_ants_spwmap=(1./45)*np.max(np.array(nflags)+np.array(nunflagged))
 
```

Comment:
I think the if statements in lines 3000 and 3005 that compare with  max_flagged_ants_combspw and  max_flagged_ants_spwmap need to have their values turned into percentages based on the the minimum number of unflagged antennas from minimum_flagged_ants_spwcomb (line 3000) and minimum_flagged_ants_spwcomb (line 3005). Otherwise those values will possibly always be larger than the percentages.

Reply:
The above two lines (2993 and 2995) are supposed to use the `2./45` and `1./45` to set the percentage of antennas flagged and then the `np.max(np.array(nflags)+np.array(nunflagged))` was a shortcut to get the total number of antennas (without having to explicitly load it in again). So I think this should give a # of antennas flagged threshold - but let me know if you see any issues with that logic, especially the second part (as I was trying to be clever and avoid loading in the antenna count again...).

Reply:
I see that makes sense, the logic escaped me there. We'll have to see how this works in the refactored code once we port the feature there.

Reply:
It may also be worth thinking about that 45 number. I initially used it as a "typical" number of ALMA antennas, but it could lead to fairly different behavior depending on the number of antennas an observation has. 43 would mean that 2 antennas would trigger, whereas 45 or more it would not trigger. Not sure whether it is desirable to have such different behavior for small changes. We could alternatively use 43 (i.e. the minimum number an ALMA observation should have)... or could do some sort of rounding to get an integer number.